lyluatex.lua

```diff
@@ -262,74 +273,40 @@ function bbox.get(filename, line_width)
-        return ps_units * 1.003749861
-    end
-
-    local function get_bb()
-        -- get "LoRes" BoundingBox from EPS file
-        local f = io.open(filename..'.eps', 'r')
-        if not f then return end
-        local bbline = ''
-        while not bbline:find('^%%%%BoundingBox') do bbline = f:read() end
-        f:close()
-        x_1, y_1, x_2, y_2 =
-            string.match(bbline, '(%--%d+)%s(%--%d+)%s(%--%d+)%s(%--%d+)')
-
-        -- try to get HiResBoundingBox from PDF (if 'gs' works)
-        local p = io.popen('gs -sDEVICE=bbox -q -dBATCH -dNOPAUSE '..filename..'.pdf 2>&1', 'r')
-        if p then
-            p:read()
-            local bbline = p:read()
-            p:close()
-            hr_x_1, hr_y_1, hr_x_2, hr_y_2 =
-                string.match(bbline,
-                  '(%--%d+%.%d+)%s(%--%d+%.%d+)%s(%--%d+%.%d+)%s(%--%d+%.%d+)')
-            -- The HiRes BoundingBox retrieved from the PDF differs from the
-            -- BoundingBox present in the EPS file. In the PDF (0|0) is the
-            -- Lower Left corner while in the EPX (0|0) represents the top
-            -- edge at the start of the staff symbol.
-            -- Therefore we shift the HiRes results by the (truncated)
-            -- points of the EPS bounding box.
-            x_1 = hr_x_1 + x_1
-            x_2 = hr_x_2 + x_1
-            y_1 = hr_y_1 + y_1
-            y_2 = hr_y_2 + y_1
-        end
-      return ps2pt(x_1), ps2pt(y_1), ps2pt(x_2), ps2pt(y_2)
+    -- get BoundingBox from EPS file
+    local bbline = readlinematching('^%%%%BoundingBox', io.open(filename..'.eps', 'r'))
+    if not bbline then return end
+    local x_1, y_1, x_2, y_2 = bbline:match('(%--%d+)%s(%--%d+)%s(%--%d+)%s(%--%d+)')
+    -- try to get HiResBoundingBox from PDF (if 'gs' works)
+    bbline = readlinematching(
+        '^%%%%HiResBoundingBox',
+        io.popen('gs -sDEVICE=bbox -q -dBATCH -dNOPAUSE '..filename..'.pdf 2>&1', 'r')
+    )
+    if bbline then
+        local pbb = bbline:gmatch('(%d+%.%d+)')
+        -- The HiRes BoundingBox retrieved from the PDF differs from the
+        -- BoundingBox present in the EPS file. In the PDF (0|0) is the
+        -- Lower Left corner while in the EPS (0|0) represents the top
+        -- edge at the start of the staff symbol.
+        -- Therefore we shift the HiRes results by the (truncated)
+        -- points of the EPS bounding box.
+        x_1, y_1, x_2, y_2 = pbb() + x_1, pbb() + y_1, pbb() + x_1, pbb() + y_1
+    else warn([[gs couldn't be launched; there could be rounding errors.]])
     end
-
-    x_1, y_1, x_2, y_2 = get_bb()
     local bb = {
-        ['protrusion'] = -x_1,
-        ['r_protrusion'] = x_2 - line_width,
-        ['width'] = x_2,
-        ['height'] = y_2 - y_1
+        ['protrusion'] = -convert_unit(x_1..'bp'),
+        ['r_protrusion'] = convert_unit(x_2..'bp') - line_width,
+        ['width'] = convert_unit(x_2..'bp'),
+        ['height'] = convert_unit(y_2..'bp') - convert_unit(y_1..'bp')
```

Comment:
Are you sure it is appropriate to exchange the PostScript Unit conversions with `bp`? They are quite close but not exactly the same.

Reply:
According to [this](https://raw.githubusercontent.com/tweh/tex-units/master/tex-units_preview.png), 1bp = 1/72in.

In fact, if I look at the result of `\directlua{print('\string\n', (tex.sp('1in')/tex.sp('1bp')) / 72)}`, I get `1.0000114014685`, which means an error of 0.0011%. On the width of an A4 sheet, this means an error of 0.0024mm.